packages/grid/x-data-grid-premium/src/hooks/features/cellSelection/useGridCellSelection.ts

```diff
@@ -235,8 +339,37 @@ export const useGridCellSelection = (
         { id, field },
         event.ctrlKey || event.metaKey,
       );
+
+      const virtualScrollerRect =
+        apiRef.current.virtualScrollerRef?.current?.getBoundingClientRect();
+
+      if (!virtualScrollerRect) {
+        return;
+      }
+
+      const { height, width, x, y } = virtualScrollerRect;
+      const mouseX = event.clientX - x;
+      const mouseY = event.clientY - y;
+      mousePosition.current = { x: mouseX, y: mouseY };
+
+      const hasEnteredVerticalSensitivityArea =
+        mouseY <= AUTO_SCROLL_SENSITIVITY || mouseY >= height - AUTO_SCROLL_SENSITIVITY;
+
+      const hasEnteredHorizontalSensitivityArea =
+        mouseX <= AUTO_SCROLL_SENSITIVITY || mouseX >= width - AUTO_SCROLL_SENSITIVITY;
+
+      const hasEnteredSensitivityArea =
+        hasEnteredVerticalSensitivityArea || hasEnteredHorizontalSensitivityArea;
+
+      if (hasEnteredSensitivityArea || !autoScrollInterval.current) {
+        // Mouse has entered the sensitity area for the first time
+        startAutoScroll();
+      } else if (!hasEnteredSensitivityArea && autoScrollInterval.current) {
+        // Mouse has left the sensitivity area while auto scroll is on
+        stopAutoScroll();
+      }
```

Comment:
Wouldn't `startAutoScroll` be called too often, even if the cursor is far from the edge of the virtual scroller?

```suggestion
      if (hasEnteredSensitivityArea) {
        // Mouse has entered the sensitity area for the first time
        startAutoScroll();
      } else {
        // Mouse has left the sensitivity area while auto scroll is on
        stopAutoScroll();
      }
```